### ISSUE TITLE
Async tests which use Assert.Inconclusive report as Error

### ISSUE THREAD

**Issue by Terebi42** [Microsoft/testfx#249]
## Description
Async tests which use Assert.Inconclusive report as Error

## Steps to reproduce
```
   [TestMethod]
        public void TestInconclusiveSync()
        {

            Assert.Inconclusive("This will report as skipped");
        }

        [TestMethod]
        public async Task TestInconclusiveAsync()
        {

            Assert.Inconclusive("This will report as error");
        }
```

v1.1.18

**Comment by AbhitejJohn** [comment]
Thanks for filing this. I can repro this. I guess this seems to be happening because the error is thrown on another thread which gets wrapped in as an internal exception we do not check.

**Comment by Terebi42** [comment]
This behavior is seen in several places, VS, TFS build, Resharper, etc.  Do you think all of those would be resolved by this bug (because they are all relying on something in the nuget), or should I be filing  bugs with those products? (because they will have to do this same exception peeking on their own)?

**Comment by AbhitejJohn** [comment]
@Terebi42 : Fixing this bug should take care of all the client scenarios.

**Comment by abatishchev** [comment]
I bugged the VSTS team with this just to learn down the road that it's unfortunately a bug in MSTest indeed. Any plans to address, and asap? It's 100% reproducible.

**Comment by AbhitejJohn** [comment]
@abatishchev: We have been busy with a few other things and this one is currently parked. I can try and help anyone from the community who is willing to take a stab at fixing this though. The code to invoke the test method is [here](https://github.com/Microsoft/testfx/blob/master/src/Adapter/MSTest.CoreAdapter/Execution/TestMethodInfo.cs#L207)

**Comment by jessehouwing** [comment]
I tried debugging this, which wasn't easy with all of the different test frameworks being loaded at the same time (sort of test framework inception). I think I may have found the issue where the Exception being thrown is from the MstestV2 test framework, but the exception being compared against is from the V1 framework. I am not a 100% sure, as my knowledge of the framework is limited.

I have forked and committed my changes here: 

https://github.com/jessehouwing/testfx

**Comment by AbhitejJohn** [comment]
@Jessehouwing: Thanks. The [change](https://github.com/jessehouwing/testfx/commit/be3187262daeb822f7b1fd99427be457a2ec943f#diff-25d902c24283ab8cfbac54dfa101ad31) looks good to me. Can you please submit a PR? 
The multiple frameworks confusion that you were running into is limited to the `Unit Test` Framework in this repo only and isn't tied to the product code. It looks like `ex is UTF.AssertInconclusiveException` did the trick.

**Comment by jthelin** [comment]
I have the same issue, so I tried the latest daily build `1.2.0-build-20170919-01` which contains the PR change #277 above, but i still have the async test case failing with error `Assert.Inconclusive failed.`

I used slightly different async test case code which included an `await` call, but it is still functionality equivalent to the example @Terebi42 gave originally.

``` csharp
[TestMethod]
public async Task Assert_Inconclusive_Async()
{
    await Task.Delay(1);

    Assert.Inconclusive("Assert_Inconclusive_Async");
}
```

Does the PR fix #277 work for async test cases for other people?

Or could a test case something like the above be added to the "official" test suite to confirm functionality?

**Comment by jessehouwing** [comment]
I'll add that locally. The test should be easy to add and check.

**Comment by jthelin** [comment]
Oops - my bad. I just noticed the "latest" build i was using from myget is `0919` [and not `0929` as i had thought]

So the real problem is that no "daily" builds have been produced in the last 10 days.

https://dotnet.myget.org/feed/mstestv2/package/nuget/MSTest.TestFramework

Could someone give the build hamsters a poke to wake them up?

**Comment by jessehouwing** [comment]
As far as I can tell your test case should succeed on the latest codebase.

**Comment by jthelin** [comment]
Thanks for confirming @jessehouwing 

I should be able to use a local dev build for the moment, when I find a machine with UWP SDK installed.... ;)

**Comment by abatishchev** [comment]
Just in case my test method is the following:
```csharp
[TestMethod]
public async Task TestAsync()
{
    await Task.Yield();

    Assert.Inconclusive();
}
```

**Comment by AbhitejJohn** [comment]
@jthelin : Sorry about that. The builds are failing because of service issues. Should be up this week hopefully.

**Comment by jthelin** [comment]
Confirming this issue is fixed for me now that v1.2.0 nuget packages published.

**Comment by jayaranigarg** [comment]
Thanks for confirming. Closing the issue.
